cli/src/program.rs

```diff
@@ -2076,23 +2123,20 @@ fn do_process_program_write_and_deploy(
     }
 }
 
+#[allow(clippy::too_many_arguments)]
 fn do_process_program_upgrade(
     rpc_client: Arc<RpcClient>,
     config: &CliConfig,
-    program_data: &[u8],
-    program_id: &Pubkey,
+    program_data: &[u8], // can be empty, hence we have program_len
+    program_len: usize,
```

Comment:
While I appreciate the fix here, it's not clear at all why this change was made, so please explain when you're making unrelated changes in the PR description, or put the change in a separate commit, or even better, make a separate PR!

Just so I'm sure I understand, the current behavior of reusing an existing buffer for a deployment will always succeed the check in  `complete_partial_program_init` of:

``` 
if !account.data.is_empty() && account.data.len() < account_data_len {
```

because `account_data_len` will be incorrectly zero, is that it?

Reply:
Ok great, thanks!